Tests/Common/Securities/Futures/FuturesExpiryFunctionsTests.cs

```diff
@@ -16,12 +16,114 @@
 using System;
 using NUnit.Framework;
 using QuantConnect.Securities.Future;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+using System.Xml.Serialization;
+using System.IO;
 
 namespace QuantConnect.Tests.Common.Securities.Futures
 {
     [TestFixture]
     public class FuturesExpiryFunctionsTests
     {
+        /// <summary>
+        /// Dates for Termination Conditions of futures
+        /// </summary>
+        public class Dates
+        {
+            public DateTime contractMonth;
+            public DateTime lastTrade;
+            public Dates() { }
+            public Dates(DateTime c, DateTime l)
+            {
+                contractMonth = c;
+                lastTrade = l;
+            }
+        }
+
+        /// <summary>
+        /// Symbol and list of dates for testing Futures
+        /// </summary>
+        public class SymbolData
+        {
+            public String symbol;
+            public List<Dates> dateList;
+            public SymbolData() { }
+            public SymbolData(String symbol, List<Dates> list)
+            {
+                this.symbol = symbol;
+                this.dateList = list;
+            }
+        }
+
+        [Test]
+        public void TestAllExpiryDateFunctions()
+        {
+            var _path = Directory.GetCurrentDirectory();
+            _path = _path.Substring(0,_path.Length-10) + "\\Common\\Securities\\Futures\\FuturesExpiryFunctionsTestData.xml";
```

Comment:
Did you set the `"Copy if newer"` flag in the new data file ?

In VS, File Properties -> Copy to output directory -> Copy if newer